utils/utils.go

```diff
@@ -325,28 +325,55 @@ func CustomizeNetworkPolicy(networkPolicy *networkingv1.NetworkPolicy, ba common
 		networkPolicy.Spec.Ingress[0].From = append(networkPolicy.Spec.Ingress[0].From, networkingv1.NetworkPolicyPeer{})
 	}
 
-	networkPolicy.Spec.Ingress[0].From[0].NamespaceSelector = &metav1.LabelSelector{}
-
-	// Customize the network policy peer
-	if exposed := ba.GetExpose(); exposed != nil && *exposed {
-		customizeExposedNetworkPolicyPeer(&networkPolicy.Spec.Ingress[0])
+	networkPolicyConfig := ba.GetNetworkPolicy()
+	if isNetworkPolicyConfigSet(networkPolicyConfig) {
+		customizeNetworkPolicyPeer(&networkPolicy.Spec.Ingress[0].From[0], networkPolicyConfig)
+	} else if isExposed := ba.GetExpose(); isExposed != nil && *isExposed {
+		customizeExposedNetworkPolicyPeer(&networkPolicy.Spec.Ingress[0].From[0], isOpenShift)
 	} else {
-		customizeNetworkPolicyPeer(networkPolicy.Namespace, &networkPolicy.Spec.Ingress[0], ba)
+		customizeInternalNetworkPolicyPeer(&networkPolicy.Spec.Ingress[0].From[0], networkPolicy.Namespace, ba)
```

Comment:
The logic in customizeNetworkPolicyPeer and customizeInternalNetworkPolicyPeer probably should be combined. - Use `"app.kubernetes.io/part-of"` as pod selector label by default. If fromLabels is specified in CR, use it instead. 
- Restrict to CR namespace by default. If namespaceLabels is specified, specify it instead.
- both can be set to `{}` to allow all pods/namespaces